### PR TITLE
Upgrade mobile to React Native 0.61.5

### PR DESCRIPTION
**Pull request by bryvin**
This change includes a lot of updates and switches to AndroidX and CocoaPods. Submodules have also been updated alongside this branch to comply with newest React Native updates.

### CODE REVIEW

File: android/app/src/main/java/com/standardnotes/MainApplication.java [hunk 1]
```diff
@@ -49,23 +49,32 @@ public boolean getUseDeveloperSupport() {
...
+      // packages.add(new MyReactNativePackage());
+
+//      packages.add(new MainReactPackage(),
+//            new RNScreensPackage(),
+//            new ReanimatedPackage());
+
+//      packages.add(new AsyncStoragePackage());
+//      packages.add(new RNCWebViewPackage());
+//      packages.add(new RNFileViewerPackage());
+//      packages.add(new RNFSPackage());
```
Comment (mobitar) [inline android/app/src/main/java/com/standardnotes/MainApplication.java]: @bryvin clean comments

Comment (bryvin) [reply android/app/src/main/java/com/standardnotes/MainApplication.java]: I removed these with the latest commit. Good call

File: ios/Podfile [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  target 'StandardNotesTests' do
+    inherit! :search_paths
+    # Pods for testing
+  end
+
+  use_native_modules!
+end
+
+target 'StandardNotes-tvOS' do
```
[the remark this answers is not in the document]
Comment (bryvin) [reply ios/Podfile]: I removed these with the latest commit. Not needed to build.

File: ios/StandardNotes/Info.plist [hunk 3]
```diff
@@ -76,6 +76,8 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
+		<key>NSAllowsArbitraryLoads</key>
```
[the remark this answers is not in the document]
Comment (bryvin) [reply ios/StandardNotes/Info.plist]: I removed this with the latest commit; after some reading it does seem it's not required and as you said would disable the security. The RN Upgrade Helper had included this for some reason.

File: src/screens/Notes/Notes.js [hunk 4]
```diff
@@ -377,13 +377,19 @@ export default class Notes extends Abstract {
   }
 
   onSearchTextChange = (text) => {
-    this.searching = true;
-    this.options.setSearchTerm(text);
+    return new Promise((resolve, reject) => {
```
Comment (mobitar) [inline src/screens/Notes/Notes.js]: @bryvin Wouldn't just marking this function as `async` accomplish the same thing? I don't think you need to manually return a promise here sync the function is synchronous. Also, what prompted the change to need to return a promise?

File: src/screens/Settings/Settings.js [hunk 5]
```diff
@@ -88,6 +88,10 @@ export default class Settings extends Abstract {
     this.mergeState({hasPasscode: hasPasscode, hasBiometrics: hasBiometrics, storageEncryption: encryptedStorage})
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    return super.shouldComponentUpdate(nextProps, nextState);
```
Comment (mobitar) [inline src/screens/Settings/Settings.js]: @bryvin if we're just calling super here, and not doing anything else, then we can just delete our subclass's implementation of this function.

Comment (bryvin) [reply src/screens/Settings/Settings.js]: Right, this was part of my potential debugging for the padding issue. Will remove.

File: src/screens/Notes/Notes.js (hunk 4 above)
Comment (bryvin) [reply src/screens/Notes/Notes.js]: These were in error; the Readme for react-native-search-box said some of the functions must return a Promise. Late night debugging is not for me anymore.

**Comment by mobitar** [comment]
Merged in https://github.com/standardnotes/mobile/commit/b922c59bc0c7e74280b065595cd576c861b3f2e2 🎉

**Pull request closed without merging**